backend/src/v2/compiler/argocompiler/argo.go

```diff
@@ -434,3 +461,69 @@ var dummyImages = map[string]bool{
 	"argostub/createpvc": true,
 	"argostub/deletepvc": true,
 }
+
+// convertStructToPVCSpec converts a protobuf Struct to a PersistentVolumeClaimSpec using JSON marshaling/unmarshalling.
+func convertStructToPVCSpec(structVal *structpb.Struct) (*k8score.PersistentVolumeClaimSpec, error) {
+	if structVal == nil {
+		return nil, nil
+	}
+	jsonBytes, err := structVal.MarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+	var pvcSpec k8score.PersistentVolumeClaimSpec
+	if err := json.Unmarshal(jsonBytes, &pvcSpec); err != nil {
+		return nil, err
+	}
+	return &pvcSpec, nil
+}
+
+// getWorkspacePVC constructs a PersistentVolumeClaim for the workspace.
+// It uses the default PVC spec (from API server config), and applies any user-specified
+// overrides from the pipeline spec, including the requested storage size.
+func getWorkspacePVC(
+	workspace *pipelinespec.WorkspaceConfig,
+	opts *Options,
+) (k8score.PersistentVolumeClaim, error) {
+	sizeStr := workspace.GetSize()
+	if sizeStr == "" && opts != nil && opts.DefaultWorkspaceSize != "" {
+		sizeStr = opts.DefaultWorkspaceSize
+	}
+	if sizeStr == "" {
+		sizeStr = fallbackWorkspaceSize
+	}
+
+	k8sWorkspace := workspace.GetKubernetes()
+	var pvcSpec k8score.PersistentVolumeClaimSpec
+	// Only use the default PVC spec when the user requests a workspace.
+	// This allows us to populate missing defaults (such as storage class, access modes, etc.)
+	// if the user did not specify them in their workspace config.
+	if opts != nil && opts.DefaultWorkspace != nil {
+		pvcSpec = *opts.DefaultWorkspace.DeepCopy()
+	}
```

Comment:
Currently it looks like the code doesn't require admins to configure workspaces to allow users to use it, I think that's fine, just want to make sure that's the intention here - if so an inline code comment indicating as such is probably warranted